portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +33,49 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+    private List<String> comments = new ArrayList<>();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        Query find = new Query("comment").addSort("timestamp", SortDirection.DESCENDING);
+        DatastoreService data = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = data.prepare(find);
+
+        //Construct an array of comment history
+        List<String> history = new ArrayList<>();
+        for (Entity entry: results.asIterable()) {
+            String curr = (String) entry.getProperty("comment");
+            history.add(curr);
+        }
+
+	    String json = convertToGson(history);
```

Comment:
fix indent

Reply:
hmm weird, I didn't have this indent in my files - maybe I edited before and forgot to push

Reply:
You are using a mix of tabs and spaces for indentation.

Tabs are discouraged because they are inconsistent. It's possible your editor is showing tabs as a different width from what GitHub shows, which would cause the indent to appear different for you locally from what it shows on github.